File: leetcode_problems/cpp/lc1614_max_nest_depth_of_parentheses.cpp
#include <string>
#include <stack>
#include <algorithm>
using std::stack;
using std::string;

class Solution
{
public:
    int maxDepth(string s)
    {
        int max = 0;
        stack<char> st;
        for (char chr : s)
        {
            if (chr == '(')
            {
                st.push(chr);
                max = std::max(max, static_cast<int>(st.size()));
            }
            else if (chr == ')')
            {
                st.pop();
            }
        }
        return max;
    }
};

File: leetcode_problems/typescript/src/lc1614_max_nest_depth_of_parentheses.ts
function maxDepth(s: string): number {
  let max = 0;
  let stack: string[] = [];
  for (let char of s) {
    if (char == "(") {
      stack.push("(");
      max = Math.max(max, stack.length);
    } else if (char == ")") {
      stack.pop();
    }
  }
  return max;
}
